src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,232 @@
+package duke.parser;
+
+import java.time.DateTimeException;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.exceptions.DukeException;
+import duke.exceptions.InvalidArgumentException;
+import duke.exceptions.InvalidCommandException;
+import duke.exceptions.MissingArgumentException;
+import duke.storage.Deadline;
+import duke.storage.Event;
+import duke.storage.Todo;
+
+/**
+ * The Parser class handles the making sense of user commands
+ *
+ * @author Ryan NgWH
+ */
+public class Parser {
+    /**
+     * Converts user input date (in format 'YYYY/MM/DD hh:mm') to Instant
+     *
+     * @param date Date to be converted (in format 'YYYY/MM/DD')
+     * @param time Time to be converted (in format 'hh:mm')
+     *
+     * @return Instant of the specified datetime
+     */
+    public static Instant userDateToInstant(String date, String time)
+            throws NumberFormatException, StringIndexOutOfBoundsException, DateTimeException {
+        return LocalDateTime.of(
```

Comment:
I think you can explore the use of the .split function to split by the '/' character. it may be less wordy + more evolvable than substring.